docs/deprecations.md

```diff
@@ -6,6 +6,12 @@ layout: Doc
 
 # Serverless Framework Deprecations
 
+<a name="PROVIDER_IAM_SETTINGS"><div>&nbsp;</div></a>
+
+## Grouping IAM settings under `provider.iam`
```

Comment:
Here it would be good to include information that the old syntax will be deprecated starting with `v3.0.0` version :+1: